### ISSUE TITLE
`graph build` significantly slower since updated to v0.22.0

### ISSUE THREAD

**Issue by schmidsi** [graphprotocol/graph-ts#212]
We noticed that `graph build` is significantly slower since v0.22.0. Does anybody else experienced this?

**Comment by otaviopace** [comment]
This is due mostly because the AssemblyScript compiler is extremely slow.

And since we jumped from the first available version in their Github (0.6) to the latest one (0.19), there were a lot of new features, ABI change, etc.

I've heard more people on our Discord complaining about this. It would be nice to have an actual time comparison between verions if you have that data 🙂 

Still, I was discussing with @leoyvens the other day about this, and there's probably room for optimization on how we compile the files nowadays, in the sense that:

- There are somethings we probably can do in parallel
- We probably can stop compiling some files if we know they've already been compiled and/or didn't change
